lib/ansible/modules/network/checkpoint/checkpoint_access_rule.py

```diff
@@ -252,14 +255,16 @@ def main():
     else:
         if code == 200:
             code, response = delete_access_rule(module, connection)
-
+            if code != 200:
```

Comment:
This will never execute because to get here `if code == 200` has to be `True`

Reply:
I agree, but this check is added only to be executed when `code != 200` i.e. failure scenarios. If `code == 200` then module shall execute as normal and return the results as expected.